www/js/survey/multilabel/MultiLabelButtonGroup.tsx

```diff
@@ -164,16 +188,47 @@ const MultilabelButtonGroup = ({ trip, buttonsInline = false }) => {
               <ScrollView style={{ paddingBottom: 24 }}>
                 <RadioButton.Group
                   onValueChange={(val) => onChooseLabel(val)}
-                  value={chosenLabel || ''}>
+                  // if 'other' button is selected and input component shows up, make 'other' radio button filled
+                  value={otherLabel !== null ? 'other' : initialLabel || ''}>
                   {modalVisibleFor &&
-                    labelOptions?.[modalVisibleFor]?.map((o, i) => (
-                      <RadioButton.Item
-                        key={i}
-                        label={o.text || labelKeyToReadable(o.value)}
-                        value={o.value}
-                        style={{ paddingVertical: 2 }}
-                      />
-                    ))}
+                    labelOptions?.[modalVisibleFor]?.map((o, i) => {
+                      const radioItemForOption = (
+                        <RadioButton.Item
+                          key={i}
+                          label={o.text || labelKeyToReadable(o.value)}
+                          value={o.value}
+                          style={{ paddingVertical: 2 }}
+                        />
+                      );
+                      /* if this is the 'other' option and there are some custom labels,
+                      show the custom labels section before 'other' */
+                      if (o.value == 'other' && customLabelMap[customLabelKeyInDatabase]?.length) {
+                        return (
+                          <>
+                            <Divider style={{ marginVertical: 10 }} />
+                            <Text
+                              style={{ fontSize: 12, color: colors.onSurface, paddingVertical: 4 }}>
+                              {(modalVisibleFor === 'MODE' ||
+                                modalVisibleFor === 'REPLACED_MODE') &&
+                                t('trip-confirm.custom-mode')}
```

Comment:
Ah I see that we do in fact use the same values for both `MODE` and `REPLACED_MODE`